arducate/src/components/Toolbar.js

```diff
@@ -16,11 +15,15 @@ import {
 import { Move, MoveDiagonal, RotateCw } from "lucide-react";
 import AssetHandler from "./AssetHandler";
 import { treeDataAtom } from "../atoms";
+import { supabase } from './supabaseClient'; // Adjust the import path as necessary
+import { v4 as uuidv4 } from 'uuid'; // Install uuid package for unique IDs
 
 const Toolbar = () => {
   const [arObjects] = useAtom(arObjectsAtom);
   const [data, setData] = useAtom(treeDataAtom);
   const [cursor, setCursor] = useState(null);
+  const [showPopup, setShowPopup] = useState(false); // State to control popup visibility
+  const [generatedUrl, setGeneratedUrl] = useState(sessionStorage.getItem('ar_content_url') || null); // Check sessionStorage for the URL
```

Comment:
why would we store the url in session storage?

Reply:
so that if u press publish or share multiple times it refers to the same sequence instead of creating a new database entry. lmk if u have a better idea